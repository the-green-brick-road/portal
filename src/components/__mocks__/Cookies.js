/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Image component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box, Typography }             from '@mui/material';

function Cookies(props) {

    /* --------- Gather inputs --------- */
    const { color } = props;

    /* -------- Defining theme --------- */
    return (
        <Box style={{ minHeight:'30px',zIndex:1000, position:'absolute', backgroundColor:color }} >
            <Typography> Cookies </Typography>
        </Box>
    );

}

export default Cookies;
