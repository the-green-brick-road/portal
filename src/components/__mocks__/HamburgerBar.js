/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# HamburgerBar component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box, Typography }   from '@mui/material';

function HamburgerBar(props) {

    /* --------- Gather inputs --------- */
    const { height, isNegative, theme={} }  = props
    /* const componentName = 'HamburgerBar'; */

    /* -------- Defining theme --------- */
    let color = theme.palette.primary.main;
    if (isNegative) { color = theme.palette.common.white; }

    return (
        <Box style={{height:height, backgroundColor:color, width: '100%'}}>
            <Typography> HamburgerBar </Typography>
        </Box>
    );

}

export default HamburgerBar;
