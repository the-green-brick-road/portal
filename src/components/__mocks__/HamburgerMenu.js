/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# HamburgerMenu component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box, Typography }   from '@mui/material';

function HamburgerMenu(props) {

    /* --------- Gather inputs --------- */
    const { top , margin, isNegative, theme={} } = props;
    /* const componentName = 'HamburgerBar'; */

    /* -------- Defining theme --------- */
    let color = theme.palette.primary.main;
    if (isNegative) { color = theme.palette.common.white; }
    const menuWidthString = `calc( 100vw - 2 * ${margin})`;

    return (
        <Box style={{ minHeight:'30px', width:menuWidthString, top:top, zIndex:1000, position:'absolute', backgroundColor:color }} >
            <Typography> HamburgerMenu </Typography>
        </Box>
    );

}

export default HamburgerMenu;