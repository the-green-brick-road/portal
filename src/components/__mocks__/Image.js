/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Image component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box }             from '@mui/material';

function Image(props) {

    /* --------- Gather inputs --------- */
    const { style={} }    = props;

    const local_style = JSON.parse(JSON.stringify(style))

    if (!('backgroundColor' in local_style)) { local_style['backgroundColor'] = 'blue' }

    /* ----------- Define HTML --------- */
    /* eslint-disable padded-blocks */
    return (
        <Box style={local_style} />
    );
    /* eslint-enable padded-blocks */

}

export default Image;
