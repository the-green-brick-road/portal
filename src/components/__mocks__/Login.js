/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Login component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @29 may 2023
# Latest revision: 29 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box }             from '@mui/material';

function Login(props) {

    /* --------- Gather inputs --------- */
    const { style={} }    = props;

    const local_style = JSON.parse(JSON.stringify(style))

    if (!('backgroundColor' in local_style)) { local_style['backgroundColor'] = 'green' }

    /* ----------- Define HTML --------- */
    /* eslint-disable padded-blocks */
    return (
        <Box style={local_style} />
    );
    /* eslint-enable padded-blocks */

}

export default Login;
