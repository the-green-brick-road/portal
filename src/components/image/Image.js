/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Image component
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @22 may 2023
# Latest revision: 14 june 2023
# ---------------------------------------------------- */

/* React includes */
import { Profiler }              from 'react';

/* Portal includes */
import { useDesign, useLogging } from '../../providers';

function Image(props) {

    /* --------- Gather inputs --------- */
    const { name, style={} }    = props;
    const { images }            = useDesign();
    const { onRender }          = useLogging();
    const componentName = 'Image';

    /* ---- Select image if loaded ----- */
    let data = {}
    if (name in images) { data = images[name] }
    const label_picture = `picture-${name}`

    /* ----------- Define HTML --------- */
    /* eslint-disable padded-blocks */
    return (
        <Profiler id={componentName} onRender={onRender}>
            {(data !== {}) && (
                <picture style={style} data-testid={label_picture}>
                    { Object.entries(data).toReversed().map((item) => {

                        let result;
                        const localType = `image/${item[0]}`;
                        const localSizes = Object.entries(item[1]);
                        let localSrcset = '';
                        for (let i = 0; i < localSizes.length; i += 1) {
                            localSrcset = `${localSrcset} ${localSizes[i][1].img} ${localSizes[i][1].width} ,`;
                        }
                        if (item[0] !== 'default') {
                            result = (<source key={item[0]} srcSet={localSrcset} type={localType} />);
                        }
                        return result;
                    })}
                    <img src={data.default} style={style} alt={name}/>
                </picture>
            )}
        </Profiler>
    );
    /* eslint-enable padded-blocks */

}

export default Image;
