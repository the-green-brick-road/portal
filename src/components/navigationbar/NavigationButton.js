/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Navigation Button styling
# -------------------------------------------------------
# NadÃ¨ge LEMPERIERE, @23 may 2023
# Latest revision: 23 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Button } from '@mui/material';
import { styled } from '@mui/system';

const NavigationButton = styled(Button)(({ col }) => ({

    color: col,
    verticalAlign: 'middle',
    borderRadius: '0',
    paddingTop: '6.5px',
    paddingLeft: '6px',
    paddingRight: '6px',
    fontSize: '14px',
    fontWeight: 'bold',
    textTransform: 'capitalize',
    span: { visibility: 'hidden' },
    '&:hover': {
        borderTopStyle: 'none',
        borderBottomStyle: 'solid',
        borderLeftStyle: 'none',
        borderRightStyle: 'none',
        borderWidth: '3px',
        transition: 'border-bottom-style 2s',
    },
    '&:hover span': { visibility: 'visible' },
    a: { fontSize: '14px' },
    '@media all and (max-device-width: 720px)': { a: { fontSize: '12px' } },
}));

export default NavigationButton;
