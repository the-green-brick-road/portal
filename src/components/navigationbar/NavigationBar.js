/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Navigation Bar component
# -------------------------------------------------------
# NadÃ¨ge LEMPERIERE, @18 may 2023
# Latest revision: 23 may 2023
# ---------------------------------------------------- */

/* React includes */
import { useRef }                                            from 'react';

/* Material UI includes */
import { ButtonGroup, MenuList, MenuItem, Link }             from '@mui/material';

/* Portal includes */
import { useLogging, useMenu }                               from '../../providers';

/* Local includes */
import NavigationButton                                      from './NavigationButton';
import NavigationLogo                                        from './NavigationLogo';
import NavigationStack                                       from './NavigationStack';

function NavigationBar(props) {

    /* --------- Gather inputs --------- */
    const { height = '20px', isNegative = false, theme={} } = props
    const { logText }                                       = useLogging();
    const { selectEntry, isItemSelected, entries}           = useMenu();
    const allRef        = useRef({});
    const componentName = 'NavigationBar';

    /* -------- Defining theme --------- */
    let stackcolor = theme.palette.primary.main;
    if (isNegative) { stackcolor = theme.palette.common.white; }

    let menucolor = theme.palette.common.white;
    if (isNegative) { menucolor = theme.palette.primary.main; }

    let style = 'solid';
    if (isNegative) { style = 'none'; }

    /* ------ Manage button click ------ */
    const handleButtonClick = (event) => {

        logText(componentName, 'debug', 'workflow', ' handleButtonClick');
        selectEntry(event.target.name, !(isItemSelected[event.target.name]) );

    };

    /* ----- Manage menu selection ----- */
    const handleClose = (event) => {

        logText(componentName, 'debug', 'workflow', ' handleClose');
        selectEntry(event.target.name, false);

    };

    /* ----------- Define HTML --------- */
    return (

        <NavigationStack
            direction="row"
            justifyContent="space-between"
            alignItems="center"
            color={stackcolor}
            padding="0px"
            height={height}
        >
            <NavigationLogo
                padding="5px"
                name='logo'
            />
            <ButtonGroup
                disableElevation={true}
                disableFocusRipple={true}
                fullWidth={true}
                disableRipple={true}
                style={{ marginLeft: '10px', marginRight: '10px', width: '100%', left: 0 }}
            >
                { entries.map((item, index) => { /* Loop on all menu content to create button */

                    const IconName = item.icon;
                    const visibility = (isItemSelected[item.id] ? 'visible' : 'hidden')
                    return (
                        <div key={item.id} style={{width:'100%'}} aria-label={item.id}>
                            <NavigationButton
                                id={item.id}
                                theme={theme}
                                name={item.id}
                                className={isItemSelected[item.id] ? 'selected' : 'nonselected'}
                                ref = {ref => allRef.current[index] = ref}
                                col={stackcolor}
                                variant="text"
                                href={ 'subitems' in item ? null : item.path }
                                target={ 'subitems' in item ? null : item.target }
                                alt={item.id}
                                startIcon={<IconName/>}
                                onClick={'subitems' in item ? handleButtonClick : handleClose}
                            >
                                {item.id}
                            </NavigationButton>
                            {('subitems' in item) && (
                                <MenuList
                                    open={isItemSelected[item.id]}
                                    style={{padding:0, borderRadius:0, visibility:visibility, position:'absolute', top:height, borderWidth:'1px', borderStyle:style, borderColor:theme.palette.primary.main}}
                                >
                                    {item.subitems.map((subitem) => {

                                        return (
                                            <MenuItem onClick={handleClose} key={subitem.id} style={{backgroundColor:menucolor, color:stackcolor, borderRadius:0}}>
                                                <Link href={subitem.path} target={subitem.target} style={{ textDecoration: 'none', color:stackcolor }}>
                                                    {subitem.id}
                                                </Link>
                                            </MenuItem>
                                        );

                                    })}
                                </MenuList>
                            )}
                        </div>
                    );

                })}
            </ButtonGroup>
        </NavigationStack>
    );

}

export default NavigationBar;
