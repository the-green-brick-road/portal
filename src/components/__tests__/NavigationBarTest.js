/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# NavigationBar component test suite
# -------------------------------------------------------
# NadÃ¨ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Test includes */
import { render, act, screen, fireEvent, prettyDOM }  from '@testing-library/react'
import { expect, test}                                from '@jest/globals';

/* Material UI includes */
import { Science }                                    from '@mui/icons-material/';

/* Component under test */
import { default as NavigationBar }                   from '../../components/navigationbar/NavigationBar';

/* Mocks includes */
/* eslint-disable jest/no-mocks-import */
import { useMenu as mockUseMenu, MenuProvider as MockMenuProvider } from '../../providers/__mocks__/MenuProvider';
import { useLogging as mockUseLogging, LoggingProvider as MockLoggingProvider } from '../../providers/__mocks__/LoggingProvider';
import { default as MockImage }   from '../../components/__mocks__/Image';
import { default as MockRouter }  from '../../containers/__mocks__/Router';
import { default as MockLayout }  from '../../containers/__mocks__/Layout';
jest.mock('../../providers', () => ({
    useLogging: (() => { return mockUseLogging(); }),
    useMenu: (()    => { return mockUseMenu(); }),
}));
jest.mock("../../components", () => ({ Image: (props) => MockImage(props) }));
jest.mock("../../containers", () => ({
    Router: (props) => MockRouter(props),
    Layout: (props) => MockLayout(props),
}));

/* eslint-enable jest/no-mocks-import */

function MockLayoutWithBar() {

    const theme = {
        "palette": {
            "common": {
                "black": "#444444",
                "white": "#555555",
            },
            "primary": { "main": "#666666" },
            "secondary" : { "main": "#777777" },
        },
        "typography" : { "button": { "fontSize": "14px" } },
    };

    return(
        <MockLayout>
            <NavigationBar height='84px' isNegative={true} theme={theme}/>
        </MockLayout>
    )

};

describe("NavigationBar component" ,() => {


    const entries = [
        {
            "id":     "item1",
            "path":   "/",
            "icon":   Science,
            "target": "_top",
        },
        {
            "id": "item2",
            "icon" : Science,
            "subitems": [
                {
                    "id": "subitem21",
                    "path": "/test1",
                    "target": "_top",
                },
                {
                    "id": "subitem22",
                    "path": "/test2",
                    "target": "_top",
                }
            ],
        }
    ];

    const theme = {
        "palette": {
            "common": {
                "black": "#444444",
                "white": "#555555",
            },
            "primary": {
                "main": "#666666",
                "light": "#666667",
                "dark": "#666668",
            },
            "secondary" : {
                "main": "#777777",
                "light": "#777778",
                "dark": "#777779",
            },
            "disabled" : {
                "main": "#888888",
                "light": "#888889",
                "dark": "#888880",
            },
            "text" : {
                "primary": "#111111",
                "secondary": "#222222",
                "disabled": "#333333",
            },
        },
        "direction": "ltr",
        "typography": {
            "h1": {
                "fontSize": "2.5rem",
                "color": "#111111",
                "textTransform": "uppercase",
                "fontFamily": "Roboto Condensed",
            },
            "h2": {
                "fontSize": "2rem",
                "color": "#222222",
                "textTransform": "lowercase",
                "fontFamily": "Roboto Condensed",
            },
            "body1": {
                "fontSize": "14px",
                "color": "black",
                "fontFamily": "Roboto",
                "position": "relative",
                "textAlign": "justify",
            },
            "button": {
                "fontSize": "14px",
                "color": "black",
                "fontFamily": "Roboto",
                "position": "relative",
                "textAlign": "justify",
            },
        },
    }

    test('Should display a navigation bar', async () => {


        const view = render(

            <div>
                <MockLoggingProvider>
                    <MockMenuProvider entries={entries}>
                        <NavigationBar height='84px' isNegative={false} theme={theme}/>
                    </MockMenuProvider>
                </MockLoggingProvider>
            </div>

        )
        const tree = prettyDOM(view.baseElement, Number.POSITIVE_INFINITY, {filterNode: () => true, escapeString: false,highlight: false});
        expect(tree).toMatchSnapshot();

    })

    test('Should display a negative navigation bar', async () => {


        const view = render(

            <div>
                <MockLoggingProvider>
                    <MockMenuProvider entries={entries}>
                        <NavigationBar height='84px' isNegative={true} theme={theme}/>
                    </MockMenuProvider>
                </MockLoggingProvider>
            </div>

        )
        const tree = prettyDOM(view.baseElement, Number.POSITIVE_INFINITY, {filterNode: () => true, escapeString: false,highlight: false});
        expect(tree).toMatchSnapshot();

    })

    test('Should select menu item on click', async () => {


        const state = mockUseMenu();

        await act(async () => { // eslint-disable-line testing-library/no-unnecessary-act

            render(

                <div>
                    <MockLoggingProvider>
                        <MockMenuProvider entries={entries}>
                            <MockRouter layout={MockLayoutWithBar} />
                        </MockMenuProvider>
                    </MockLoggingProvider>
                </div>

            )

        })


        const delay = ms => new Promise(res => setTimeout(res, ms));
        await delay(1000);

        await act(async () => {fireEvent.click(screen.getByRole('link', { name: 'item1' }))}) // eslint-disable-line testing-library/no-unnecessary-act
        expect(state.selectEntry).toHaveBeenLastCalledWith('item1', false)
        await act(async () => {fireEvent.click(screen.getByRole('button', { name: 'item2' }))}) // eslint-disable-line testing-library/no-unnecessary-act
        expect(state.selectEntry).toHaveBeenLastCalledWith('item2', true)

        /* eslint-disable testing-library/no-node-access*/
        expect(screen.getByText('item1').closest('a').getAttribute('href')).toBe('/')
        expect(screen.getByText('subitem21').closest('a').getAttribute('href')).toBe('/test1')
        expect(screen.getByText('subitem22').closest('a').getAttribute('href')).toBe('/test2')
        /* eslint-enable testing-library/no-node-access*/


    })

})
