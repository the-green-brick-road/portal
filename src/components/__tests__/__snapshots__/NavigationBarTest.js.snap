// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar component Should display a navigation bar 1`] = `
"<body>
  <div>
    <div>
      <div
        class=\\"MuiStack-root css-fzqd5a-MuiStack-root\\"
      >
        <a
          class=\\"MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1645xoy-MuiTypography-root-MuiLink-root\\"
          href=\\"/\\"
          logopadding=\\"5px\\"
        >
          <div
            class=\\"MuiBox-root css-0\\"
            style=\\"height: 100%; vertical-align: middle; background-color: blue;\\"
          />
        </a>
        <div
          class=\\"MuiButtonGroup-root MuiButtonGroup-outlined MuiButtonGroup-fullWidth MuiButtonGroup-disableElevation css-8q4ur5-MuiButtonGroup-root\\"
          role=\\"group\\"
          style=\\"margin-left: 10px; margin-right: 10px; width: 100%; left: 0px;\\"
        >
          <div
            aria-label=\\"item1\\"
            style=\\"width: 100%;\\"
          >
            <a
              alt=\\"item1\\"
              class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButton-fullWidth MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButton-fullWidth nonselected css-xg3fdv-MuiButtonBase-root-MuiButton-root\\"
              col=\\"#666666\\"
              href=\\"/\\"
              id=\\"item1\\"
              name=\\"item1\\"
              tabindex=\\"0\\"
              target=\\"_top\\"
            >
              <span
                class=\\"MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon\\"
              >
                <svg
                  aria-hidden=\\"true\\"
                  class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\"
                  data-testid=\\"ScienceIcon\\"
                  focusable=\\"false\\"
                  viewBox=\\"0 0 24 24\\"
                >
                  <path
                    d=\\"M19.8 18.4 14 10.67V6.5l1.35-1.69c.26-.33.03-.81-.39-.81H9.04c-.42 0-.65.48-.39.81L10 6.5v4.17L4.2 18.4c-.49.66-.02 1.6.8 1.6h14c.82 0 1.29-.94.8-1.6z\\"
                  />
                </svg>
              </span>
              item1
            </a>
          </div>
          <div
            aria-label=\\"item2\\"
            style=\\"width: 100%;\\"
          >
            <button
              alt=\\"item2\\"
              class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButton-fullWidth MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButton-fullWidth nonselected css-xg3fdv-MuiButtonBase-root-MuiButton-root\\"
              col=\\"#666666\\"
              id=\\"item2\\"
              name=\\"item2\\"
              tabindex=\\"0\\"
              type=\\"button\\"
            >
              <span
                class=\\"MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon\\"
              >
                <svg
                  aria-hidden=\\"true\\"
                  class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\"
                  data-testid=\\"ScienceIcon\\"
                  focusable=\\"false\\"
                  viewBox=\\"0 0 24 24\\"
                >
                  <path
                    d=\\"M19.8 18.4 14 10.67V6.5l1.35-1.69c.26-.33.03-.81-.39-.81H9.04c-.42 0-.65.48-.39.81L10 6.5v4.17L4.2 18.4c-.49.66-.02 1.6.8 1.6h14c.82 0 1.29-.94.8-1.6z\\"
                  />
                </svg>
              </span>
              item2
            </button>
            <ul
              class=\\"MuiList-root MuiList-padding css-h4y409-MuiList-root\\"
              role=\\"menu\\"
              style=\\"padding: 0px; border-radius: 0; visibility: hidden; position: absolute; top: 84px; border-width: 1px; border-style: solid; border-color: #666666;\\"
              tabindex=\\"-1\\"
            >
              <li
                class=\\"MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root\\"
                role=\\"menuitem\\"
                style=\\"background-color: rgb(85, 85, 85); color: rgb(102, 102, 102); border-radius: 0;\\"
                tabindex=\\"0\\"
              >
                <a
                  class=\\"MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1bhi0za-MuiTypography-root-MuiLink-root\\"
                  href=\\"/test1\\"
                  style=\\"text-decoration: none; color: rgb(102, 102, 102);\\"
                  target=\\"_top\\"
                >
                  subitem21
                </a>
                <span
                  class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"
                />
              </li>
              <li
                class=\\"MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root\\"
                role=\\"menuitem\\"
                style=\\"background-color: rgb(85, 85, 85); color: rgb(102, 102, 102); border-radius: 0;\\"
                tabindex=\\"-1\\"
              >
                <a
                  class=\\"MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1bhi0za-MuiTypography-root-MuiLink-root\\"
                  href=\\"/test2\\"
                  style=\\"text-decoration: none; color: rgb(102, 102, 102);\\"
                  target=\\"_top\\"
                >
                  subitem22
                </a>
                <span
                  class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"
                />
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>"
`;

exports[`NavigationBar component Should display a negative navigation bar 1`] = `
"<body>
  <div>
    <div>
      <div
        class=\\"MuiStack-root css-10auekq-MuiStack-root\\"
      >
        <a
          class=\\"MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1645xoy-MuiTypography-root-MuiLink-root\\"
          href=\\"/\\"
          logopadding=\\"5px\\"
        >
          <div
            class=\\"MuiBox-root css-0\\"
            style=\\"height: 100%; vertical-align: middle; background-color: blue;\\"
          />
        </a>
        <div
          class=\\"MuiButtonGroup-root MuiButtonGroup-outlined MuiButtonGroup-fullWidth MuiButtonGroup-disableElevation css-8q4ur5-MuiButtonGroup-root\\"
          role=\\"group\\"
          style=\\"margin-left: 10px; margin-right: 10px; width: 100%; left: 0px;\\"
        >
          <div
            aria-label=\\"item1\\"
            style=\\"width: 100%;\\"
          >
            <a
              alt=\\"item1\\"
              class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButton-fullWidth MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButton-fullWidth nonselected css-1jecpqq-MuiButtonBase-root-MuiButton-root\\"
              col=\\"#555555\\"
              href=\\"/\\"
              id=\\"item1\\"
              name=\\"item1\\"
              tabindex=\\"0\\"
              target=\\"_top\\"
            >
              <span
                class=\\"MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon\\"
              >
                <svg
                  aria-hidden=\\"true\\"
                  class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\"
                  data-testid=\\"ScienceIcon\\"
                  focusable=\\"false\\"
                  viewBox=\\"0 0 24 24\\"
                >
                  <path
                    d=\\"M19.8 18.4 14 10.67V6.5l1.35-1.69c.26-.33.03-.81-.39-.81H9.04c-.42 0-.65.48-.39.81L10 6.5v4.17L4.2 18.4c-.49.66-.02 1.6.8 1.6h14c.82 0 1.29-.94.8-1.6z\\"
                  />
                </svg>
              </span>
              item1
            </a>
          </div>
          <div
            aria-label=\\"item2\\"
            style=\\"width: 100%;\\"
          >
            <button
              alt=\\"item2\\"
              class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButton-fullWidth MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButton-fullWidth nonselected css-1jecpqq-MuiButtonBase-root-MuiButton-root\\"
              col=\\"#555555\\"
              id=\\"item2\\"
              name=\\"item2\\"
              tabindex=\\"0\\"
              type=\\"button\\"
            >
              <span
                class=\\"MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon\\"
              >
                <svg
                  aria-hidden=\\"true\\"
                  class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\"
                  data-testid=\\"ScienceIcon\\"
                  focusable=\\"false\\"
                  viewBox=\\"0 0 24 24\\"
                >
                  <path
                    d=\\"M19.8 18.4 14 10.67V6.5l1.35-1.69c.26-.33.03-.81-.39-.81H9.04c-.42 0-.65.48-.39.81L10 6.5v4.17L4.2 18.4c-.49.66-.02 1.6.8 1.6h14c.82 0 1.29-.94.8-1.6z\\"
                  />
                </svg>
              </span>
              item2
            </button>
            <ul
              class=\\"MuiList-root MuiList-padding css-h4y409-MuiList-root\\"
              role=\\"menu\\"
              style=\\"padding: 0px; border-radius: 0; visibility: hidden; position: absolute; top: 84px; border-width: 1px; border-style: none; border-color: #666666;\\"
              tabindex=\\"-1\\"
            >
              <li
                class=\\"MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root\\"
                role=\\"menuitem\\"
                style=\\"background-color: rgb(102, 102, 102); color: rgb(85, 85, 85); border-radius: 0;\\"
                tabindex=\\"0\\"
              >
                <a
                  class=\\"MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1bhi0za-MuiTypography-root-MuiLink-root\\"
                  href=\\"/test1\\"
                  style=\\"text-decoration: none; color: rgb(85, 85, 85);\\"
                  target=\\"_top\\"
                >
                  subitem21
                </a>
                <span
                  class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"
                />
              </li>
              <li
                class=\\"MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root\\"
                role=\\"menuitem\\"
                style=\\"background-color: rgb(102, 102, 102); color: rgb(85, 85, 85); border-radius: 0;\\"
                tabindex=\\"-1\\"
              >
                <a
                  class=\\"MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1bhi0za-MuiTypography-root-MuiLink-root\\"
                  href=\\"/test2\\"
                  style=\\"text-decoration: none; color: rgb(85, 85, 85);\\"
                  target=\\"_top\\"
                >
                  subitem22
                </a>
                <span
                  class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"
                />
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>"
`;
